bot/role_management/role_management.py

```diff
@@ -131,6 +130,203 @@ async def module_role_error(self, ctx: commands.Context, error: commands.Command
+                           "\"exklusiv\".")
+        else:
+            self._db_connector.add_reaction_role_group(message.id)
+            await ctx.send(":white_check_mark: Die Reaction-Roles der angegebenen Nachricht sind nun \"exklusiv\".")
+
+    @add_reaction_role.error
+    @remove_reaction_role.error
+    @clear_reaction_roles.error
+    @toggle_reaction_roles_exclusiveness.error
+    async def reaction_role_error(self, _ctx: commands.Context, error: commands.CommandError):
+        """Error Handler for the `reactionrole` command group.
+
+        Handles specific exceptions which occur during the execution of this command. The global error handler will
+        still be called for every error thrown.
+
+        Args:
+            _ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error, commands.BadArgument):
+            print("**__Error:__** Die von dir angegebene Nachricht/Rolle existiert nicht.")
+
+    @commands.Cog.listener(name='on_raw_reaction_add')
+    async def reaction_role_add(self, payload: discord.RawReactionActionEvent):
+        """Event listener which triggers if a reaction has been added by a user.
+
+        If the affected message is in the specified role channel and the added reaction represents one of the configured
+        reaction roles, the corresponding role specified in the db will be added to the user.
+
+        Args:
+            payload (discord.RawReactionActionEvent): The payload for the triggered event.
+        """
+        if payload.channel_id == self.ch_role.id and not payload.member.bot:
+            if self._db_connector.check_reaction_role_uniqueness(payload.message_id):
+                message = await self.ch_role.fetch_message(payload.message_id)
+
+                for reaction in message.reactions:
+                    if reaction.emoji != payload.emoji.name:
+                        await reaction.remove(payload.member)
+
+            role_id = self._db_connector.get_reaction_role(payload.message_id, payload.emoji.name)
+            role = self.bot.get_guild(payload.guild_id).get_role(role_id)
+            await payload.member.add_roles(role, reason="Selbstzuweisung via Reaction.")
+
+    @commands.Cog.listener(name='on_raw_reaction_remove')
+    async def reaction_role_remove(self, payload: discord.RawReactionActionEvent):
+        """Event listener which triggers if a reaction has been removed.
+
+        If the affected message is in the specified role channel and the removed reaction represents one of the
+        configured reaction roles, the corresponding role specified in the db will removed from the user.
+
+        Args:
+            payload (discord.RawReactionActionEvent): The payload for the triggered event.
+        """
+        role_id = self._db_connector.get_reaction_role(payload.message_id, payload.emoji.name)
```

Comment:
Same here, what if someone adds and removes a reaction that does not have a associated role?

Reply:
An AttributeError happens which will then be silently ignored. I mean, I could catch it and do something with it but why bother? Generally users shouldn't be able to add new reactions to messages in the roles channel and if a member still manages to achieve this in some mysterious way, nothing is going to happen.

Reply:
Can they really not?

Well it polutes the logs...

Reply:
Yeah, because they don't have the permission to do this in this channel.

That's true, but so does every other exception as of right now. We could change that in the future somehow.